src/io/stm_tests.ml

```diff
@@ -0,0 +1,298 @@
+open QCheck
+open STM
+
+(** parallel STM tests of Out_channels *)
+
+module OCConf =
+struct
+  type cmd =
+    | Open_text
+    | Seek of int64
+    | Pos
+    | Length
+    | Close
+    | Close_noerr
+    | Flush
+    | Output_char of char
+    | Output_byte of int
+    | Output_string of string
+    | Output_bytes of bytes
+    | Output of bytes * int * int
+    | Output_substring of string * int * int
+    | Set_binary_mode of bool
+    | Set_buffered of bool
+    | Is_buffered
+
+  let pp_cmd par fmt x =
+    let open Util.Pp in
+    match x with
+    | Open_text -> cst0 "Open_text" fmt
+    | Seek i -> cst1 pp_int64 "Seek" par fmt i
+    | Pos -> cst0 "Pos" fmt
+    | Length -> cst0 "Length" fmt
+    | Close -> cst0 "Close" fmt
+    | Close_noerr -> cst0 "Close_noerr" fmt
+    | Flush -> cst0 "Flush" fmt
+    | Output_char c -> cst1 pp_char "Output_char" par fmt c
+    | Output_byte i -> cst1 pp_int "Output_byte" par fmt i
+    | Output_string s -> cst1 pp_string "Output_string" par fmt s
+    | Output_bytes b -> cst1 pp_bytes "Output_bytes" par fmt b
+    | Output (b,p,l) -> cst3 pp_bytes pp_int pp_int "Output" par fmt b p l
+    | Output_substring (s,p,l) -> cst3 pp_string pp_int pp_int "Output_substring" par fmt s p l
+    | Set_binary_mode b -> cst1 pp_bool "Set_binary_mode" par fmt b
+    | Set_buffered b -> cst1 pp_bool "Set_buffered" par fmt b
+    | Is_buffered -> cst0 "Is_buffered" fmt
+
+  let show_cmd = Util.Pp.to_show pp_cmd
+
+  (* a path and an open channel to that file; we need to keep the path
+     to cleanup after the test run *)
+  type sut = { path            : string;
+               mutable channel : Out_channel.t }
+
+  type state = Closed of int64
```

Comment:
I don’t think I’ve seen the argument of `Closed` used anywhere. Did I miss it?

Reply:
Indeed! I think this is a left-over from when I started. 
Initially I think I allowed all `cmd`s in both states and tried to track length for closed `Out_channel`s too.
Fixed in df541cc